contracts/HtsSystemContract.sol

```diff
@@ -940,4 +954,14 @@ contract HtsSystemContract is IHederaTokenService {
         assembly { sstore(slot, approved) }
         emit IERC721.ApprovalForAll(sender, operator, approved);
     }
+
+    function _keyExists(uint keyType, TokenInfo memory tokenInfo) private pure returns (bool) {
+        for (uint256 i = 0; i < tokenInfo.token.tokenKeys.length; i++) {
+            if (tokenInfo.token.tokenKeys[i].keyType == keyType) {
```

Comment:
`keyType` is a bit field, so we need to check if the right bit is set. However, we can leave it for another PR.